aws-android-sdk-s3/src/main/java/com/amazonaws/mobileconnectors/s3/transferutility/TransferRecord.java

```diff
@@ -364,9 +364,8 @@ private boolean checkPreferredNetworkAvailability(final TransferStatusUpdater up
             // Unable to get the details of the network, we will start the transfer.
             return true;
         }
-            
-        if (transferUtilityOptions != null &&
-            !transferUtilityOptions.getTransferNetworkConnectionType().isConnected(connManager)) {
+
+        if (transferUtilityOptions != null && transferUtilityOptions.getTransferNetworkConnectionType() != null && !transferUtilityOptions.getTransferNetworkConnectionType().isConnected(connManager)) {
```

Comment:
Added null check will skip updating the state and return true for case when 
`transferUtilityOptions.getTransferNetworkConnectionType()` is null
is it expected?

Reply:
I think it would be a good idea to add a unit test on it as well.

Reply:
Just returning without updating the state to WAITING_FOR_NETWORK might have unexpected outcome because TransferNetworkLossHandler queues all the tasks marked with WAITING_FOR_NETWORK when the network is available again. And also isConnected is deprecated and not reliable starting Android O. 

That said it is ok to skip updating the state incase of null to prevent the app from crashing.

Reply:
> Added null check will skip updating the state and return true for case when `transferUtilityOptions.getTransferNetworkConnectionType()` is null is it expected?

Yes as this will only return false when it is null and at that time the app will just crash

Reply:
> Just returning without updating the state to WAITING_FOR_NETWORK might have unexpected outcome because TransferNetworkLossHandler queues all the tasks marked with WAITING_FOR_NETWORK when the network is available again. And also isConnected is deprecated and not reliable starting Android O.
> 
> That said it is ok to skip updating the state incase of null to prevent the app from crashing.

Yeah that code is still unchanged just added an extra null guard. Regarding the deprecated aspect since we support Jelly Bean and up we have to keep this code.